queries.md

```diff
@@ -4,76 +4,133 @@
+query: { $and: [ { founded_year: { $gte: 2000 } }, { founded_year: { $lte: 2005 } } ] }
+projection: { name: 1, founded_year: 1, _id: 0 }
+
+<!-- db.companies.find({ $and: [ { founded_year: { $gte: 2000 } }, { founded_year: { $lte: 2005 } } ] }, { name: 1, founded_year: 1, _id: 0 }).pretty(); -->
 
 ### 4. All the companies that had a Valuation Amount of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
 
-<!-- Your Code Goes Here -->
+query: { $and: [ { "ipo.valuation_amount": { $gt: 100000000 } }, { founded_year: { $lt: 2010 } } ] }
+projection: { name: 1, ipo: 1, _id: 0 }
+
+<!-- db.companies.find({ $and: [ { "ipo.valuation_amount": { $gt: 100000000 } }, { founded_year: { $lt: 2010 } } ] }, { name: 1, ipo: 1, _id: 0 }).pretty(); -->
 
 ### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
 
-<!-- Your Code Goes Here -->
+query: { $and: [ { number_of_employees: { $lt: 1000 } }, { founded_year: { $lt: 2005 } } ] }
+sort: { number_of_employees: 1 }
+limit: (10)
+
+<!-- db.companies.find({ $and: [ { number_of_employees: { $lt: 1000 } }, { founded_year: { $lt: 2005 } } ] }).sort({ number_of_employees: 1 }).limit(10).pretty(); -->
 
 ### 6. All the companies that don't include the `partners` field.
 
-<!-- Your Code Goes Here -->
+query: { partners: { $size: 0 } }
+
+<!-- db.companies.find({ partners: { $size: 0 } }).pretty(); -->
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
-<!-- Your Code Goes Here -->
+query: { category_code: { $type: 'null' } }
+
+<!-- db.companies.find({ category_code: { $type: 'null' } }).pretty(); -->
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
-<!-- Your Code Goes Here -->
+query: { $and: [ { number_of_employees: { $gte: 100 } }, { number_of_employees: { $lt: 1000 } } ] }
+projection: { "name":1, "number_of_employees":1, _id:0 }
+
+<!-- db.companies.find({ $and: [ { number_of_employees: { $gte: 100 } }, { number_of_employees: { $lt: 1000 } } ] }, { "name":1, "number_of_employees":1, _id:0 }).pretty(); -->
 
 ### 9. Order all the companies by their IPO price in a descending order.
 
-<!-- Your Code Goes Here -->
+sort: { "ipo.valuation_amount": -1 }
+
+<!-- db.companies.find().sort({"ipo.valuation_amount":-1}).pretty(); -->
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+sort: { number_of_employees: -1 }
+limit: (10)
+
+<!-- db.companies.find().sort({ number_of_employees: -1 }).limit(10).pretty(); -->
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+query: { $and: [ {founded_month: { $gt:6 } }, { founded_month: { $lte:12 } } ] }
```

Comment:
i think that the second case is not really necessary. by default if its bigger than 6, it will reach the numbers up to 12, so I dont believe you needed the second case here